### PR TITLE
Add GetReservationOrder and GetReservation in new 2020-11-01-preview [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by toddm**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [X] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [X] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by asarkar84** [comment]
Do we have these all APIs working in the new version?
If not then we should not create a PR with all APIs.

**Comment by chiragg4u** [comment]
While adding new version the recommendation is to copy the existing version into new directory structure for first commit and then push new changes for easy review. Please create new branch with this recommendation for faster review.
In this case looks like you've renamed the file, please help us identify the changes so that we can review the changes more efficiently.

**Comment by toddm** [comment]
In this case, there are 4 new APIs in the new preview directory and they are all new.  It does not contain a copy of the existing version.  My team doesn't even own all of the APIs in the stable version so we're not in a position to make them all available in the preview version.
So the reservations.json file is all new.

**Comment by chiragg4u** [comment]
Looks like these API's are exposing completely new scenarios, which requires a discussion with ARM API Review board. In case you've already meet with the team, please share the notes. Otherwise please schedule meeting with the  ARM API review board oncall next week to get closure on this.

**Comment by ravbhatnagar** [comment]
GET reservationOrders API is also present on Microsoft.Capacity RP. The response is different currently. The team has ackd that they will fix the capacity RP response as well in the next api version. 
Other than that, the APIs look fine. They are adding APIs to get reservation Orders and reservations under billingAccount and billing profile.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/ReservationOrderByBillingAccount.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          "expiryDate": null,
+          "reviewDate": null,
+          "endDate": null,
+          "term": "None",
+          "reservationName": "SQL_Server_Enterprise_Edition",
+          "billingScope": "",
+          "billingPlanDetails": null,
+          "state": "Created",
+          "subState": "Active",
+          "reservations": [
```
Comment (ravbhatnagar) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/ReservationOrderByBillingAccount.json:27]: do you really need to surface reservations in the response of reservationOrders? The customer can do an explicit GET to get all reservations under a reservationOrder.

Comment (toddm) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/ReservationOrderByBillingAccount.json:27]: We did the same thing in Microsoft.Capacity for the reservationOrder. https://github.com/Azure/azure-rest-api-specs/blob/e0b78897850ccbb648b3efe286b78e78c5cd8bcf/specification/reservations/resource-manager/Microsoft.Capacity/stable/2017-11-01/reservations.json#L737

**Pull request closed without merging**